ThePhasesOfCraftship/2_best_practice_first/strategicDesignPart1/exercises/1_RefactoringTo4Tiers/begin/tests/e2e/assignStudentToAssignment.test.ts

```diff
@@ -0,0 +1,120 @@
+import { defineFeature, DefineStepFunction, loadFeature } from "jest-cucumber";
+import path from "path";
+import supertest from "supertest";
+import request from "supertest";
+import { app, Errors } from "../../src";
+import { resetDatabase, AssignmentBuilder, StudentBuilder } from "../fixtures";
+import { Assignment, Student } from "@prisma/client";
+
+const feature = loadFeature(
+  path.join(__dirname, "../features/assignStudentToAssignment.feature")
+);
+
+defineFeature(feature, (test) => {
+
+    beforeEach(resetDatabase);
+
+    afterAll(() => {
+        app.close();
+    });
+
+    let student: Student;
+    let assignment: Assignment;
+
+    const givenIHaveAStundentAndAnAssignment = ({given, and}:{given: DefineStepFunction, and: DefineStepFunction}) => {
+        given('I have a student', async() => {
+            student = await StudentBuilder.Fake().build();
+        });
+
+        and('I have an assignment', async () => {
+            assignment = await AssignmentBuilder.Fake().build();
+        });
+    }
```

Comment:
I'm just speaking my opinion :) but I think is an unnecessary abstraction. For me it's clearer if you put it directly under the test below

Reply:
What do you mean exactly? The usage of builders or extracting the `givenIHaveAStundentAndAnAssignment` method?

If it's the latter I stole the idea from the jest-cucumber repo. I was looking for a way to properly implement the gherkin `Background` statements, and this is the closest I got :D 

References:
https://github.com/bencompton/jest-cucumber/blob/main/examples/typescript/specs/features/backgrounds.feature
https://github.com/bencompton/jest-cucumber/blob/main/examples/typescript/specs/step-definitions/backgrounds.steps.ts

Reply:
I was referencing the `givenIHaveAStundentAndAnAssignment ` but now I see it from the docs that you provided that it makes sense